### PR TITLE
Prepare release of version 0.8.0

### LINKED ISSUE #550: Prepare a new release (version 0.8.0)
There have been multiple requests for a new rp2040-hal release, and the last one was 2 months ago, so we should prepare a new one soon.

As some changes were, at least formally, breaking ones (eg. added new functions to public traits), the version will be 0.8.0.

The following pull requests should be merged (or rejected / deferred, if necessary), first:
- https://github.com/rp-rs/rp-hal/pull/541 (has been ready for a while, but still needs a review)
- https://github.com/rp-rs/rp-hal/pull/549 (quite new, but could avoid some semver breaking changes in future releases)

All other pull requests are either in draft state, or need some updates before they can be merged.

While there are a lot of open issues (37 at the moment), I think none of them is serious enough to count as a blocker. In fact, most are feature requests.

### PR DESCRIPTION
**Pull request by jannic**
Closes #550

### CODE REVIEW

**Comment by 9names** [comment]
Should we also close #13 with this release, or should that wait until we have DMA support for all peripherals as originally stated?

File: rp2040-hal/CHANGELOG.md [hunk 1]
```diff
@@ -237,7 +239,8 @@ The Minimum-Supported Rust Version (MSRV) for this release is 1.54.
 
 - Initial release
 
-[Unreleased]: https://github.com/rp-rs/rp-hal/compare/v0.7.0...HEAD
+[Unreleased]: https://github.com/rp-rs/rp-hal/compare/v0.8.0...HEAD
+[0.7.0]: https://github.com/rp-rs/rp-hal/compare/v0.7.0...v0.8.0
 [0.7.0]: https://github.com/rp-rs/rp-hal/compare/v0.6.0...v0.7.0
```
Comment (9names) [inline rp2040-hal/CHANGELOG.md:244]:
```suggestion
[0.8.0]: https://github.com/rp-rs/rp-hal/compare/v0.7.0...v0.8.0
[0.7.0]: https://github.com/rp-rs/rp-hal/compare/v0.6.0...v0.7.0
```

**Review by 9names — APPROVED** [review]
LGTM

File: rp2040-hal/CHANGELOG.md (hunk 1 above)
Comment (jannic) [reply rp2040-hal/CHANGELOG.md:244]: LOL, thanks!

**Pull request merged**